### PR TITLE
Add HashedModuleIdsPlugin to webpack definitions

### LINKED ISSUE #18701: webpack is missing HashedModuleIdsPlugin
- [x] I tried using the @types/react-native package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for StackOverflow. (Please ask any appropriate questions there).
- [x] Mention the authors (see Definitions by: in index.d.ts) so they can respond.
Authors: @tkqubo @bumbleblym @bcherny @tommytroylin @mohsen1

The exports on the webpack object does not have HashedModuleIdsPlugin so when trying to run webpack production builds with ts-node, it won't compile.

### PR DESCRIPTION
**Pull request by jcreamer898**
Closes #18701 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/webpack/index.d.ts [hunk 1]
```diff
@@ -832,6 +832,14 @@ declare namespace webpack {
         constructor();
     }
 
+    class HashedModuleIdsPlugin extends Plugin {
+        constructor(options?: {
+            hashFunction: string,
```
Comment (mohsen1) [inline types/webpack/index.d.ts]: Are all these properties required?

**Review by mohsen1 — CHANGES_REQUESTED** [review]
Fix linting and indentation issues

File: types/webpack/index.d.ts (hunk 1 above)
Comment (jcreamer898) [inline types/webpack/index.d.ts]: Nope, ha, fixing.

**Comment by jcreamer898** [comment]
Link to docs... https://webpack.js.org/plugins/hashed-module-ids-plugin/

**Review by mohsen1 — APPROVED** [review]

**Pull request merged**